lib/hello/catalog/product.ex

```diff
@@ -0,0 +1,26 @@
+defmodule Hello.Catalog.Product do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  alias Hello.Catalog.Category
+
+
+  schema "products" do
+    field :description, :string
+    field :price, :decimal
+    field :title, :string
+    field :views, :integer
+
+    #join products with categories
+    many_to_many :categories, Category, join_through: "product_categories", on_replace: :delete
+
+    timestamps()
+  end
+
+  @doc false
+  def changeset(product, attrs) do
+    product
+    |> cast(attrs, [:title, :description, :price, :views])
```

Comment:
@keliumJU could you please explain this line of code?

Reply:
the cast is a function that is responsible for determining which parameters of a structure will have validations, in this case, it would read something like this: of all product attributes(attrs), these [:title, :description, :price, :views] will have validations